stylix/target.nix

```
@@ -31,18 +31,28 @@
     };
   };
 
-  config.lib.stylix.mkEnableTarget = let
+  config.lib.stylix = let
     cfg = config.stylix;
-  in
-    humanName:
-    autoEnable:
-      lib.mkEnableOption
-      "theming for ${humanName}"
-      // {
-        default = cfg.enable && cfg.autoEnable && autoEnable;
-        example = !autoEnable;
-      }
-      // lib.optionalAttrs autoEnable {
-        defaultText = lib.literalMD "same as [`stylix.autoEnable`](#stylixautoenable)";
-      };
+  in {
+    mkEnableTarget = 
+      humanName:
+      autoEnable:
+        lib.mkEnableOption
+        "theming for ${humanName}"
+        // {
+          default = cfg.enable && cfg.autoEnable && autoEnable;
+          example = !autoEnable;
+        }
+        // lib.optionalAttrs autoEnable {
+          defaultText = lib.literalMD "same as [`stylix.autoEnable`](#stylixautoenable)";
+        };
+    mkEnableWallpaper = 
+      humanName:
+        lib.mkEnableOption
+        "toggle for wallpaper for ${humanName}"
+        // {
+          default = config.stylix.image != null;
+          example = config.stylix.image == null;
+        };
+    };
```

This will result in `"Whether to enable toggle for wallpaper for ${humanName}."`

We might want to replace the initial "enable" by re-implementing `lib.mkEnableOption`:

```suggestion
    mkEnableWallpaper = humanName:
      mkOption {
        default = config.stylix.image != null;
        example = config.stylix.image == null;
        description = "Whether to set the wallpaper for ${humanName}.";
        type = lib.types.bool;
      };
```

---

reimplimented as a hybrid of suggested change and mkEnableTarget